src/main/java/org/opensearch/neuralsearch/processor/rerank/context/QueryContextSourceFetcher.java

```diff
@@ -107,4 +124,11 @@ private static Map<String, Object> requestToMap(final SearchRequest request) thr
         Map<String, Object> map = parser.map();
         return map;
     }
+
+    private boolean validatePath(final String path) {
```

Comment:
can we add a validation on a string length, something like >= 1000 characters is not valid? With current logic we first split the string into tokens, then validate. If there is in malicious input such operation may cause overflow before being stopped by a validator